incubator/hnc/internal/validators/object.go

```diff
@@ -111,18 +112,32 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	// If the object wasn't and isn't inherited, we will check to see if the
 	// source can be created without causing any conflict.
 	if !oldInherited && !newInherited {
+		// check selector format
+		oldSelectorStr, _ := selectors.GetSelectorAnnotation(oldInst)
+		newSelectorStr, _ := selectors.GetSelectorAnnotation(inst)
+		sel, err := selectors.GetSelector(log, inst)
+		// If this is a selector change, and the new selector is not valid, we'll deny this operation
+		if oldSelectorStr != newSelectorStr && sel == nil {
+			msg := fmt.Sprintf("Invalid %s value: %q is not a valid Kubernetes labelSelector: \n  %s", api.AnnotationSelector, newSelectorStr, err)
```

Comment:
We usually don't put newlines in our validation error messages, so while it's a good idea, I think I'd take this out.

Also, if you follow my advice above about how to wrap the error message, this message can be much shorter since more of the information will already be in the error, e.g.:

```
msg := fmt.Sprintf("Invalid Kubernetes labelSelector: %s", err")
```

Reply:
... also note the message that you're currently displaying:

```
The request is invalid: : Invalid propagate.hnc.x-k8s.io/select value: "cqoornttdrkjacuiqagaqottaprzpjzfjudvlqziugeywbzgbdfvcxcimwiecodcpehldykmrikoxixv" is not a valid Kubernetes labelSelector:
  couldn't parse the selector string "cqoornttdrkjacuiqagaqottaprzpjzfjudvlqziugeywbzgbdfvcxcimwiecodcpehldykmrikoxixv": unable to parse requirement: invalid label key "cqoornttdrkjacuiqagaqottaprzpjzfjudvlqziugeywbzgbdfvcxcimwiecodcpehldykmrikoxixv": name part must be no more than 63 characters
```

There's a _lot_ of repeated information in that message. What do you think it _should_ look like?

Reply:
The part 
```
couldn't parse the selector string 
"cqoornttdrkjacuiqagaqottaprzpjzfjudvlqziugeywbzgbdfvcxcimwiecodcpehldykmrikoxixv":
unable to parse requirement: invalid label key
"cqoornttdrkjacuiqagaqottaprzpjzfjudvlqziugeywbzgbdfvcxcimwiecodcpehldykmrikoxixv":
name part must be no more than 63 characters
```
comes from k8s. I'm thinking keeping it and just add `Invalid Kubernetes labelSelector: ` in the front. Basically what you suggested above. Wdyt?

Reply:
That sgtm!

Reply:
done